src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.cs

```diff
@@ -216,10 +216,17 @@ private static void GenerateWriteableField(ArrayBuilder<CodeAction> result, Sema
                 if (containingMethod.Parameters.Length > 0)
                 {
                     var compilation = await document.Project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
-                    var lastParameterIsCancellationToken = Equals(containingMethod.Parameters.Last().Type, compilation.CancellationTokenType());
 
-                    if (lastParameterIsCancellationToken)
-                        parameterIndex--;
+                    for (var i = containingMethod.Parameters.Length - 1; i >= 0; i--)
+                    {
+                        var parameter = containingMethod.Parameters[i];
+                        if (parameter.HasExplicitDefaultValue
+                            || Equals(parameter.Type, compilation.CancellationTokenType())
```

Comment:
This doesn't check if the parameter is last. Is that a deliberate change?

Also, what about out parameters?

Also, because I've seen bugs in this area before, is there a test for extension methods to skip over the first `this` parameter? What about inserting into `public bool IsSilly(this CancellationToken token) => true;`?

Reply:
it goes backwards through the parameters until it finds the insertion position, instead of just looking at the very last parameter.

I did not know there were any rules about ordering for out parameters. As for skipping over the first this, that;s a great question. I'd need to test it out to see what it does currently

Reply:
It's not a rule, but certainly there is a convention that out parameters come last. I don't honestly know what the convention is for a method with both out parameters and a cancellation token.

Reply:
We're inconsistent in roslyn. 

I think my preference would be CT, then out params. But I don't feel strongly.

Reply:
Should we try to insert the parameter before out parameters? There are no language rules about it so someone could have
`public void Method(out int x, int y, out int z)`
Then it becomes a question of, do we want to insert it before the last out parameter or the first one. At that point, I think we should just insert into the first available position that will not create errors.

Reply:
we probably want to place before all out-params and CT if those are at the end of the method.  in other words, keep walking backwards as long as we see something at teh end of hte list that likely should stay at the end of hte list.